dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -81,24 +94,64 @@ object Datadog {
                 InternalLogger.Target.USER,
                 CANNOT_CREATE_SDK_INSTANCE_ID_ERROR
             )
-            return
+            return null
         }
 
-        globalSdkCore = DatadogCore(context, credentials, configuration, sdkInstanceId)
-        globalSdkCore.setTrackingConsent(trackingConsent)
+        val sdkCore = DatadogCore(context, credentials, configuration, sdkInstanceId)
+        sdkCore.setTrackingConsent(trackingConsent)
+        registry.register(instanceName, sdkCore)
```

Comment:
should we make a double-check (synchronise it and do `getInstance` check again) here to protect from concurrency issues? It is possible that 2 threads are calling `initialize` method at the same time with same instance name, both go through initial `getInstance` check and come to this point, creating 2 `DatadogCore` instances, but only one of them will be in registry in the end (but both will be returned). wdyt?